packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -503,6 +594,107 @@ test.serial('writeGranulesFromMessage() saves the same values to DynamoDB, Postg
   t.deepEqual(omit(translatedPgRecord, omitList), omit(esRecord, omitList));
 });
 
+test.serial('writeGranulesFromMessage() given a partial granule updates only provided fields', async (t) => {
+  const {
+    collectionCumulusId,
+    cumulusMessage,
+    esGranulesClient,
+    granuleModel,
+    granulePgModel,
+    knex,
+    executionCumulusId,
+    providerCumulusId,
+    granuleId,
+  } = t.context;
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  const originalpgGranule = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+
+  const { updatedPgGranuleFields, pgGranule } = await updateGranule(t, true);
+
+  // FUTURE:
+  // 1. 'created_at' is updated during PUT/PATCH
+  // 2. 'published' defaults to false if not provided in the payload
+  // 3. 'duration' comes from the workflow and will be reset on update
+  // 4. 'product_volume' comes from a files object on the payload, which may not exist
+  //   in the case of partial granule updates
```

Comment:
These are all of the cases that I found that likely need fixing in the follow-on work for PUT/PATCH.

Reply:
Maybe? It depends what you mean by "this" in "this does the expected thing with those values". We're testing `writeGranulesFromMessage` and, from that perspective, the logic for `created_at`, `published`, `duration`, and `product_volume` isn't right. If we're saying that these fields will never be a problem because the message will always have them, that's different and I think `writeGranulesFromMessage` should validate that those fields are present in the payload. 

The new tests here for `writeGranulesFromMessage` are basically saying "I should be able to provide any combination of fields on a granule as the payload and the function should work". I'd argue that right now it _doesn't_ work with partial granule payloads if e.g. I don't provide `files` in the payload and `product_volume` is updated to 0 on the granule.